modules/cli/cmds/apply.go

```diff
@@ -58,6 +62,27 @@ func (a *Apply) Run(cmd *cobra.Command, args []string) error {
 		SyncGeneration:  int64(a.SyncGeneration),
 	}
 
+	if a.UsernameFile != "" && a.PasswordFile != "" {
+		username, err := ioutil.ReadFile(a.UsernameFile)
+		if err != nil && !os.IsNotExist(err) {
+			return err
+		}
+		password, err := ioutil.ReadFile(a.PasswordFile)
+		if err != nil && !os.IsNotExist(err) {
+			return err
+		}
+
+		opts.Auth.Username = string(username)
+		opts.Auth.Password = string(password)
+	}
+	if a.CABundleFile != "" {
+		cabundle, err := ioutil.ReadFile(a.CABundleFile)
+		if err != nil && !os.IsNotExist(err) {
+			return err
+		}
+		opts.Auth.CABundle = cabundle
+	}
+
```

Comment:
So if the user provides a file, but the error is not a DNE error, then we ignore the error? Will that always be okay?

Source: https://golang.org/src/os/file.go?s=20199:20241#L641
(in Go 1.16 `ioutil.ReadFile` calls `os.ReadFile`)

Reply:
Yeah this is a concern but this is mostly used internally. The reason I ignore the error is that https://github.com/rancher/fleet/pull/321/files#diff-e45faf37c46946875b36e69c4720af10dbe9f75f5a1d8295b79347149d14205fR573-R579 this is hardcoded so it could be a case where file doesn't exist(we mount the secret into the pod and if key doesn't exist then file won't be there)